cogs/bumps.py

```diff
@@ -0,0 +1,228 @@
+            name='bump',
+            dm_permission=False,
+            default_member_permissions=Permissions(manage_guild=True)
+    )
+    async def bump(self, itx: Interaction):
+        """
+        Base slash command for bumps.
+        """
+
+
+    @bump.subcommand(name='toggle', description='Toggle the playback of bumps.')
+    async def bump_toggle(
+        self,
+        itx: Interaction,
+        toggle: bool = SlashOption(
+            name='toggle',
+            description='Turn bumps on or off?',
+            required=False
+        )
+    ):
+        """
+        Subcommand for toggling bumps.
+        """
+        if itx.guild is None:
+            raise RuntimeError('[bump::toggle] itx.guild is None')
+
+        if toggle is None:
+            enabled = self._bot.database.get_bumps_enabled(itx.guild.id)
+            status = "Bump playback is currently enabled." if enabled \
+                else "Bump playback is currently disabled."
+            return await itx.response.send_message(
+                embed=create_success_embed(
+                    title="Bumps status",
+                    body=status,
+                )
+            )
+
+        self._bot.database.set_bumps_enabled(itx.guild.id, toggle)
+        status = "Bump playback has been enabled." if toggle \
+            else "Bump playback has been disabled."
+        return await itx.response.send_message(
+            embed=create_success_embed(
+                title="Bumps toggled",
+                body=status,
+            )
+        )
+
+    @bump.subcommand(name='add', description='Add a bump.')
+    async def bump_add(
+        self,
+        itx: Interaction,
+        title: str = SlashOption(name='title', description='Title of bump.', required=True),
+        author: str = SlashOption(name='author', description='Author of bump.', required=True),
+        url: str = SlashOption(name='url', description='URL to add.', required=True),
+    ):
+        """
+        Subcommand for adding a bump.
+        """
+        if itx.guild is None:
+            raise RuntimeError('[bump::add] itx.guild is None')
+
+        if len(title) > 32 or len(author) > 32:
+            return await itx.response.send_message(
+                embed=create_error_embed(
+                    message='Titles/authors cannot exceed 32 characters in length.'
+                )
+            )
+
+        if not check_url(url):
+            return await itx.response.send_message(
+                embed=create_error_embed(
+                    message='The given URL is not valid.'
+                )
+            )
+
+        bump = self._bot.database.get_bump_by_url(itx.guild.id, url)
+        if bump is not None:
+            return await itx.response.send_message(
+                embed=create_error_embed(
+                    message='A bump with the given URL already exists.'
+                )
+            )
+
+        self._bot.database.set_bump(itx.guild.id, url, title, author)
```

Comment:
This method could be called `add_bump` instead, so it is clearer that there can be multiple bumps allowed in the database per guild.